tests/test_ampOffset.py

```diff
@@ -0,0 +1,110 @@
+# This file is part of ip_isr.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+import unittest
+import numpy as np
+
+import lsst.utils.tests
+import lsst.afw.image as afwImage
+from lsst.obs.lsst import LsstCam
+from lsst.obs.subaru import HyperSuprimeCam
+from lsst.ip.isr.ampOffset import AmpOffsetConfig, AmpOffsetTask
+
+
+class AmpOffsetTest(lsst.utils.tests.TestCase):
+    def setUp(self):
+        self.camera = None  # `lsst.afw.cameraGeom.Camera` object
+        self.backgroundFractionSample = None  # not used for now
+        self.ampEdgeWindow = None  # not used for now
+
+    def tearDown(self):
+        del self.camera
+
+    def buildExposure(self, addBackground=False):
+        exp = afwImage.ExposureF(self.camera[0].getBBox())
+        exp.setDetector(self.camera[0])
+        amps = exp.getDetector().getAmplifiers()
+        self.values = np.linspace(-2.5, 2.5, len(amps))
+        for i, amp in enumerate(amps):
+            exp.image[amp.getBBox()] = self.values[i]
+        if addBackground:
+            exp.image.array += 100
+        return exp
+
+    def testAmpOffset(self):
+        if self.camera is None:
+            raise unittest.SkipTest("Skipping due to no camera set.")
+        exp = self.buildExposure(addBackground=False)
+        amps = exp.getDetector().getAmplifiers()
+        config = AmpOffsetConfig()
+        config.doBackground = False
+        config.doDetection = False
+        task = AmpOffsetTask(config=config)
+        task.run(exp)
```

Comment:
If this is changed to return a struct, it can be used here to directly compare the values without needing to go through the metadata object.

Reply:
Done!